### ISSUE TITLE
Question regarding mac lock functionality

### ISSUE THREAD

**Issue by dcraig-sage** [oliverschwendener/ueli#157]
Hi, I have recently switched from Alfred to Ueli, one thing that irks me slightly is the difference in the lock command as this is something I use quite a lot.

with alfred lock takes me to a screen with my user avatar and I can unlock from that screen immediately using my fingerprint (or click and type my password if I wanted to).

with ueli I have to type in my username and my password again, it resumes the same session, but it feels more like a switch user than a simple lock.

if it's possible the functionality as it is in alfred would be preferred, not sure if this could be a config or if there is any suggestion for this?

Thanks

**Comment by oliverschwendener** [comment]
I know what you mean. Ueli uses commandline tools to lock your computer. The only way I could get it to work is with this script:

```
/System/Library/CoreServices/Menu\\ Extras/User.menu/Contents/Resources/CGSession -suspend
```

If you could find out how to lock your mac from the commandline to "proper" I would be happy to know :)

**Comment by oliverschwendener** [comment]
Ok I probably found the solution with this script:

```
osascript -e 'tell application "System Events" to keystroke "q" using {control down, command down}'
```

You can try it by pasting this command into Terminal. Is this what you are hoping for?

**Comment by dcraig-sage** [comment]
Closest I can find after a quick search is

pmset displaysleepnow

seems to do roughly equivalent, however this requires keyboard/mouse interaction to unsleep before seeing the user screen again - I think that personally I would still prefer this than having to type out my username again.

**Comment by dcraig-sage** [comment]
I just saw your message after I had posted mine, yes the command you linked seems perfect, exactly as it works in alfred :)

**Comment by oliverschwendener** [comment]
This fix will be included in the next major release.

**Comment by oliverschwendener** [comment]
This is included in the newest [v8.0.0](https://github.com/oliverschwendener/ueli/releases/v8.0.0) release.
